### DISCUSSION TITLE
getEntranceFee Return not matching up

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1247] [Q&A] [answered]
So I went over this a few times, double checked the video and caught some errors, but when I run getEntranceFee() fund_and_withdrawy.py my number returns 2000000 which isn't even close to rate.

fund_and_withdraw.py

```Python
from brownie import FundMe
from scripts.helpful_scripts import get_account


def fund():
    fund_me = FundMe[-1]
    account = get_account()
    entrance_fee = fund_me.getEntranceFee()
    print(entrance_fee)


def main():
    fund()
```

getEntranceFee (which I have looked at multiple times, i did manually enter it but it seems right
```Solidity
    function getEntranceFee() public view returns (uint256) {
        //minimum USD
        uint256 minimumUSD = 50 * 10**18;
        uint256 price = getPrice();
        uint256 precision = 1 * 10**18;
        return (minimumUSD * precision) / price;
    }
```
even will include my helpful_scripts.py

```Python
from lib2to3.pgen2.token import STAR
from brownie import network, config, accounts, MockV3Aggregator
from web3 import Web3

LOCAL_BLOCKCHAIN_ENVIRONMENT = ["development", "ganache-local"]
DECIMALS = 8
STARTING_PRICE = 200000000000


def get_account():
    if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENT:
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])


def deploy_mocks():
    print(f"THe active network is {network.show_active()}")
    print(f"Deploying Mocks...")
    if len(MockV3Aggregator) <= 0:
        MockV3Aggregator.deploy(DECIMALS, STARTING_PRICE, {"from": get_account()})
    print("Mocks Deployed")
```

Deploying works, everything works but when i `print(entrance_fee)` it returns 2000000

Now i spent a day trying to understand all the big numbers back when we did this on remix but now the number is too small?  Anyone see anything I did wrong?

**Comment by JohnEMagee** [comment, marked as answer]
So I still was getting aggravated over this issue.  I tried a bunch of things and realized if the issue was in MockV3Aggregator I would have to recompile it.  I tried a variety of attempts including deleting my ganache-local, shutting down ganache, restarting it and spinning up a new ganache-local add...this did not work, it kept looking for the original contract address of MockV3Aggregatory (which I verified from the map.json file.  I attempted to move that reference from map.json - but that did not work...

The solution that made this work was deleting the 1337 file.  Then running deploy, both contracts were deployed poroperly and whatever issue I was having before changed and I get the right entry fee.

I'm not sure what the problem was but this did fix it.

**Reply by cromewar** [reply]
This is really weird, as there must have being a cache error on your brownie or ganache, thanks for sharing the solution. You are awesome.

**Reply by fsegall** [reply]
I had the exact same problem. Thanks for solving the issue! For some reason, my fund and withdraw test still doesn't pass when using mainnet-fork-dev. It complains about fund_me.getEntranceFee() even if I just print it prior to sending the transaction.

**Reply by fsegall** [reply]
This is the test output:

    def test_can_fund_and_withdraw():
        account = get_account()
        fund_me = deploy_fund_me()
>       print(fund_me.getEntranceFee())
E       brownie.exceptions.VirtualMachineError: revert

**Reply by cromewar** [reply]
@fsegall it seems that your transaction is getting reverted, could you share the complete erro log please?

**Reply by fsegall** [reply]
@cromewar. I ran the test again without the print statement and it passed on all networks. It was indeed getting reverted. Thanks anyway!

...@...:~/..../brownie_fund_me/src$ brownie test --network mainnet-fork-dev
Brownie v1.18.1 - Python development framework for Ethereum

======================================================================= test session starts ========================================================================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /..../blockchainPy/brownie_fund_me/src
plugins: eth-brownie-1.18.1, hypothesis-6.27.3, forked-1.4.0, xdist-1.34.0, web3-5.27.0
collected 2 items                                                                                                                                                  

Launching 'ganache-cli --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/HjbcXY16Er-u_1K9dvZiJoTlF4JBLF8H --mnemonic brownie --port 8545 --hardfork istanbul'...

tests/test_fund_me.py .s                                                                                                                                     [100%]

=================================================================== 1 passed, 1 skipped in 9.40s ===================================================================

**Reply by cromewar** [reply]
Interesting, if you want to go in deep on this could you share the code of the getEntranceFee? and also share which ones were reverting?

**Reply by fsegall** [reply]
Thanks, @cromewar. I think I'll leave it as it is for now. 
Here's my repo, though: https://github.com/fsegall/fund_me
It has some updates to the FundMe.sol in order to work with pragma solidity >=0.8.0 <0.9.0.

**Reply by fulltimecrypto** [reply]
> This is really weird, as there must have being a cache error on your brownie or ganache, thanks for sharing the solution. You are awesome.

Interesting, I had the exact same issue as @JohnEMagee. Thanks for letting us know your solution, it fixed it on my side too.
